optimum/intel/openvino/modeling_sam.py

```diff
@@ -0,0 +1,409 @@
+            prompt_encoder_model = cls._compile_model(
+                file_names["prompt_encoder_mask_decoder_model"],
+                kwargs.get("device", "CPU"),
+                kwargs.get("ov_config"),
+                model_save_dir,
+            )
+
+        model = cls(
+            vision_encoder_model=vision_encoder_model,
+            prompt_encoder_mask_decoder_model=prompt_encoder_model,
+            config=config,
+            model_save_dir=model_save_dir,
+            **kwargs,
+        )
+
+        return model
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        use_cache: bool = True,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ):
+        compile_only = kwargs.pop("compile_only", False)
+        if compile_only:
+            logger.warning(
+                "`compile_only` mode will be disabled because it does not support model export."
+                "Please provide openvino model obtained using optimum-cli or saved on disk using `save_pretrained`"
+            )
+            compile_only = False
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
+
+        if task is None:
+            task = cls.export_feature
+
+        variant = kwargs.pop("variant", None)
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task=task,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            token=token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            stateful=False,
+            variant=variant,
+        )
+        config = AutoConfig.from_pretrained(save_dir_path, trust_remote_code=trust_remote_code)
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            use_cache=use_cache,
+            **kwargs,
+        )
+
+    @property
+    def _component_names(self):
+        base_components = ["vision_encoder", "prompt_encoder_mask_decoder"]
+        return base_components
+
+    @property
+    def components(self):
+        return {component_name: getattr(self, component_name) for component_name in self._component_names}
```

Comment:
sam is a transformers model, we only added the components property is for diffusion pipelines no ?

Reply:
how is that connected? no, components property used for correct getting models in multi-model pipelines in weights compression case, it does not depends from which library used for export, it is needed to make compression process model class independent

Reply:
😅I remember adding it first time in the diffusion refactoring, it seems you added it later to vlms at some point, we can make it a standard in the lib, but for now it's far from it, also it's not used for compression 🙂 https://github.com/search?q=repo%3Ahuggingface%2Foptimum-intel+%22components%22&type=code

Reply:
compression used ov_submodels property starting from this commit https://github.com/huggingface/optimum-intel/pull/1177, I thought that components takes part for its preparation
ok if you insist I can remove components

Reply:
no worries, I'm not insisting, I think it's neither bad not good, but I don't see the point of adding it as a public property if it doesn't serve an actual purpose.